File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int l = 0;
        int r = nums.size()-1;
        int pos = nums.size()-1;
        vector<int> result(nums.size());

        while (l <= r)
        {
            int left = abs(nums[l]);
            int right = abs(nums[r]);
            
            if (left > right)
            {
                result[pos--] = left*left;
                l++;
            }
            else{
                result[pos--] = right*right;
                r--;
            }
        }
        return result;
    }
};

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int n = nums.length;
        int[] result = new int[n];

        int left = 0;
        int right = n-1;
        int fill = n-1;

        while (left <= right)
        {
            int leftVal = Math.abs(nums[left]);
            int rightVal = Math.abs(nums[right]);

            if (leftVal > rightVal)
            {
                result[fill] = leftVal*leftVal;
                left++;
                fill--;
            } else {
                result[fill] = rightVal*rightVal;
                right--;
                fill--;
            }
        }
        return result;
    }
}